specs/interop.md

```diff
@@ -0,0 +1,540 @@
+| {0x760ee04d}      | bytes4                  | 0-3                     | n/a     |
+| baseFeeScalar     | uint32                  | 4-7                     | 1       |
+| blobBaseFeeScalar | uint32                  | 8-11                    |         |
+| sequenceNumber    | uint64                  | 12-19                   |         |
+| l1BlockTimestamp  | uint64                  | 20-27                   |         |
+| l1BlockNumber     | uint64                  | 28-35                   |         |
+| basefee           | uint256                 | 36-67                   | 2       |
+| blobBaseFee       | uint256                 | 68-99                   | 3       |
+| l1BlockHash       | bytes32                 | 100-131                 | 4       |
+| batcherHash       | bytes32                 | 132-163                 | 5       |
+| interopSetSize    | uint8                   | 164-165                 | 6       |
+| chainIds          | [interopSetSize]uint256 | 165-(32*interopSetSize) | 6+      |
+
+## Fault Proof
+
+TODO: need more detail
+
+No changes to the dispute game bisection are required. The only changes required are to the fault proof program itself.
+The insight is that the fault proof program can be a superset of the state transition function.
+
+## Sequencer Policy
+
+The sequencer can include executing transactions that have corresponding initiating transactions
+that only have preconfirmation levels of security if they trust the destination sequencer. Using an
+allowlist and identity turns sequencing into an interated game which increases the ability for
+sequencers to trust each other. Better preconfirmation technology will help to scale the sequencer
+set to untrusted actors.
+
+## Block Building
+
+The block builder SHOULD use static analysis on executing messages to verify that the initiating
+message exists. When a transaction has a top level [to][tx-to] field that is equal to the `CrossL2Inbox`
+and the 4byte selector in the calldata matches the `relayMessage(bytes32,CrossChainMessage)` interface
+the block builder should use the chain id that is encoded in the `CrossChainMessage` to know which chain includes the initiating
+transaction and then use the `_hash` to populate an `eth_getLogs` query to check for the existence of the initiating message.
+
+[tx-to]: https://github.com/ethereum/execution-specs/blob/1fed0c0074f9d6aab3861057e1924411948dc50b/src/ethereum/frontier/fork_types.py#L52
+
+The following JSON body can be used with a query to `eth_getLogs` to check the existence of an initiating message.
+The `$HASH` should be filled in with the cross domain message hash.
+
+```json
+[
+  {
+    "address": "0x4200000000000000000000000000000000000023",
+    "topics":["0x52f9c5a044b4fc98e5bcc3f7790fc36930ad9e20cbee36bfaa0232b016c925b2", "$HASH"]
+  }
+]
+```
+
+## Sponsorship
+
+If a user does not have ether to pay for the gas of an executing message, application layer sponsorship solutions can be created.
```

Comment:
Note: this applies our learning from the L1 to L2 deposit gas market, which is one of the most complex portions of our code. 

A user of this protocol currently has to pay for gas on both chains, which can be abstracted away in the app layer.